### PR TITLE
[AJDA-2893] Include datatype length for TIMESTAMP/TIME columns in table manifest

### PR DESCRIPTION
**Pull request by ondrajodas**
## Problem

When exporting a table, the generated manifest included `KBC.datatype.length` for text and numeric columns but **not** for TIMESTAMP/TIME columns, even though Snowflake types like `TIMESTAMP_NTZ(9)` carry a defined fractional-seconds precision. The custom-query export path (`getColumnsInfo()`) already produced the length, so manifests differed between the two export paths for the same column.

## Root cause

`SnowflakeMetadataProvider::processColumnData()` derived the column length only from `CHARACTER_MAXIMUM_LENGTH` (text) and `NUMERIC_PRECISION`/`NUMERIC_SCALE` (numeric). For TIMESTAMP/TIME columns both are `NULL`; Snowflake stores the precision in `DATETIME_PRECISION`, which the code never read.

## Fix

Derive `KBC.datatype.length` from `information_schema.columns.DATETIME_PRECISION` for `TIMESTAMP*`/`TIME` columns. `DATE` is deliberately excluded — it has no precision and `php-datatypes` rejects a length for it. This makes the table-export path consistent with the custom-query path.

## Changes

- Extract the length-derivation logic from `processColumnData()` into a pure, unit-testable helper `SnowflakeUtils::getColumnLength()`, adding the `DATETIME_PRECISION` fallback restricted via `isTemporalTypeWithPrecision()`.
- Add DB-free unit tests covering text, numeric, all `TIMESTAMP*` variants, `TIME`, `DATE` (excluded), zero precision, and zero character length.
- Update the 10 affected expected functional manifests (`in.c-main.auto-increment-timestamp.csv.gz.manifest`) — the `datetime` (TIMESTAMP_NTZ) column now carries `KBC.datatype.length: "9"`.

Behavior for text and numeric columns is unchanged.

Linear: AJDA-2893

### CODE REVIEW

File: src/Extractor/SnowflakeUtils.php [hunk 1]
```diff
@@ -14,4 +14,36 @@ public static function parseTypeAndLength(string $rawType): array
...
+        }
+
+        if (!is_null($column['DATETIME_PRECISION'])
+            && self::isTemporalTypeWithPrecision((string) $column['DATA_TYPE'])
+        ) {
+            return (string) $column['DATETIME_PRECISION'];
+        }
+
+        return null;
+    }
```
[the remark this answers is not in the document]
Comment (ondrajodas) [reply src/Extractor/SnowflakeUtils.php:41]: Thanks for the suggestion. This is a deliberate choice: casting the DB-`mixed` values to string and adding a single baseline entry keeps `getColumnLength()` consistent with this repo's established convention for `information_schema` row values (the baseline already carries e.g. `Cannot cast mixed to int` for the same reason). Introducing `is_numeric`/`is_string` narrowing here would be inconsistent with the surrounding code, so we're keeping it as is.

File: phpstan-baseline.neon [hunk 2]
```diff
@@ -35,6 +35,11 @@ parameters:
 			count: 1
 			path: src/Extractor/SnowflakeMetadataProvider.php
 
+		-
+			message: "#^Cannot cast mixed to string\\.$#"
+			count: 5
+			path: src/Extractor/SnowflakeUtils.php
+
```
[the remark this answers is not in the document]
Comment (ondrajodas) [reply phpstan-baseline.neon:42]: Thanks. We're keeping this baseline entry intentionally (see the reply on `SnowflakeUtils.php`). The mixed-to-string casts mirror the repo's existing convention for values read from `information_schema`, and this single new entry sits alongside the pre-existing `Cannot cast mixed to int` baseline. Not removing it.

File: src/Extractor/SnowflakeUtils.php [hunk 3]
```diff
@@ -14,4 +14,36 @@ public static function parseTypeAndLength(string $rawType): array
         $length = $m[2] ?? null;
         return [$type, $length];
     }
+
+    /**
+     * @param array<string, mixed> $column
+     */
+    public static function getColumnLength(array $column): ?string
+    {
+        if (!empty($column['CHARACTER_MAXIMUM_LENGTH'])) {
```
Comment (odinuv) [inline src/Extractor/SnowflakeUtils.php:23]: Tady mě trošku zaujalo, že je tu empty a u ostatních is_null, předpokládám, že 0 jako délka stejně není povolená, takže je to asi jedno, pokud to teda nemá nějakej skrytej význam

Comment (odinuv) [reply src/Extractor/SnowflakeUtils.php:23]: aha, tak koukám, že délka 0 je asi možná

File: src/Extractor/SnowflakeMetadataProvider.php [hunk 4]
```diff
@@ -195,18 +195,10 @@ private function queryPrimaryKeys(string $schema, string $tableName): array
...
-                $length = $column['NUMERIC_PRECISION'] . ',' . $column['NUMERIC_SCALE'];
-            } else {
-                $length = $column['NUMERIC_PRECISION'];
-            }
-        }
         $columnBuilder
             ->setName($column['COLUMN_NAME'])
             ->setDefault($column['COLUMN_DEFAULT'])
-            ->setLength($length)
+            ->setLength(SnowflakeUtils::getColumnLength($column))
```
Comment (odinuv) [inline src/Extractor/SnowflakeMetadataProvider.php:201]: A nejde to nějak sjednotit s SnowflakeUtils::parseTypeAndLength  ?

**Review by odinuv — APPROVED** [review]
asi jo

File: src/Extractor/SnowflakeUtils.php (hunk 3 above)
Comment (ondrajodas) [reply src/Extractor/SnowflakeUtils.php:23]: no nula je možná u DATETIME_PRECISION... tady mi to nedává smysl protože `varchar(0)` není přeci string 🤷‍♂️

File: src/Extractor/SnowflakeMetadataProvider.php (hunk 4 above)
Comment (ondrajodas) [reply src/Extractor/SnowflakeMetadataProvider.php:201]: asi nějak možná by to šlo, ale nedělal bych to parseTypeAndLength parsuje třeba `TIMESTAMP_NTZ(9)` na type a length a getColumnLength získává jen legth podle typu sloupečku z resultu

jsou to 2 rozdílné věci

File: src/Extractor/SnowflakeUtils.php (hunk 3 above)
Comment (odinuv) [reply src/Extractor/SnowflakeUtils.php:23]: Eh https://github.com/keboola/db-extractor-snowflake/blob/9d6e1a8cd4286828cb91fadecae2f9a85d7fc55c/src/Extractor/SnowflakeMetadataProvider.php#L101

Comment (ondrajodas) [reply src/Extractor/SnowflakeUtils.php:23]: image
